docs/spec/spec.md

```diff
@@ -3647,3 +3746,141 @@ isolated function followersLoaderFunction(readonly & anydata[] ids) returns Foll
 ```
 
 The above example utilizes three DataLoader instances: `postsLoader`, `rePostsLoader`, and `followersLoader`. These DataLoaders are associated with the batch load functions `postsLoaderFunction`, `rePostsLoaderFunction`, and `followersLoaderFunction`. The 'post' field in the example utilizes the `postsLoader` and `rePostsLoader` DataLoaders, while the 'followers' field utilizes the `followersLoader` DataLoader. This demonstrates how different fields can utilize specific DataLoaders to efficiently load and retrieve related data in GraphQL resolvers.
+
+### 10.7 Caching
+
+This section describes the caching mechanisms in the Ballerina GraphQL module.
+
+#### 10.7.1 Server-side Caching
+
+The Ballerina GraphQL module offers built-in server-side caching for GraphQL `query` operations. The caching operates as in-memory caching, implemented using the Ballerina cache module. The GraphQL module generates cache keys based on the arguments and the path. In server-side caching, the `errors` and `null` values are skipped when caching. There are two different ways called `operation caching` and `field caching` to enable server-side caching.
```

Comment:
How does the pluggable cache work here?

Reply:
Since we are yet implement the distributed caching, it's not added to the spec.

Reply:
I think it is quite straightforward. We have to have a separate field to provide `cache:Cache` object with a custom implementation. Will it take more time to implement this? If that's not the case, I'd add that feature here as well.

Reply:
But we focused on in-memory caching with the proposal, which added distributed caching as a non-goal. It may takes time and we can handle it separately.